Expensify/App
# Add login validation route back in NewDot

cc @NikkiWines 

### Details
Adds a login validation route that was removed in [this PR](https://github.com/Expensify/App/pull/5736), but should not have been.

### Fixed Issues
$ https://github.com/Expensify/App/issues/6136

### Tests
1. Log in to a validated account without any secondary logins in NewDot
1. Navigate to Settings > Profile and add a secondary login for an account that you have access to
1. Confirm the sent link routes you to NewDot not OldDot
1. Click on the link and confirm your secondary login is validated
1. Repeat steps 1 and 2
1. Click on Resend to resend the validation link to yourself
1. Repeat steps 3 and 4

### QA Steps
1. Complete *Test* steps while logged into an account on NewDot.
    - For step 5 ensure you are logged into the account when you click on the link
1. Complete *Test* steps while logged into an account on NewDot.
    - For step 5 ensure you are logged out of the account when you click on the link

### Tested On
